src/main/java/com/example/demo/matcher/services/OrderService.java

```diff
@@ -1,23 +1,103 @@
 package com.example.demo.matcher.services;
 
 import com.example.demo.matcher.models.Order;
+import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.OrderbookItem;
 
-import java.util.ArrayList;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class OrderService {
-    private ArrayList<Order> orders;
+    private final List<Order> orders;
 
-    public ArrayList<Order> get() {
+    public OrderService() {
+        orders = new ArrayList<>();
+    }
+
+    public List<Order> get() {
         return orders;
     }
 
     public void add(Order order) {
-
+        orders.add(order);
     }
 
     public void remove(Order order) {
+        orders.remove(order);
+    }
 
+    public List<OrderbookItem> getOrderbook(OrderAction action, String accountId) {
+        // filter the order list by action
+        List<Order> filtered = this.get().stream()
```

Comment:
I don't think you have to call `this.get()` instead you can just do your stream on `orders` directly. I'm assuming the getter will be useful for other classes trying to access the orders from this service.

Reply:
I'm using get so that I can mock the return value for testing. For example the test ItShouldGenerateAnOrderbookWithBuyAction(), I'm mocking the orderService.get() value on line 70

Reply:
Is this good practise?

Reply:
Ideally we wouldn't do this but yeah for the mocking this makes it easier. I think this is fine to leave 👍